azerothcore/azerothcore-wotlk
# fix(Scripts/Gossip): Trial of the crusader

## Changes Proposed:
-  Select the gossip and npc text from the database, instead of always displaying the same gossip and hardcore texts.
-  Write the npc script to look like the blizzlike content.

<details>
<summary>Before</summary>

![toc_default](https://user-images.githubusercontent.com/2810187/178842162-c167368d-cd35-42a1-aaa0-4268762ea061.png)

</details>

<details>
<summary>After</summary>

![toc_1](https://user-images.githubusercontent.com/2810187/178842315-bd280612-8926-4465-b510-c4bf648fa354.png)

![toc_2](https://user-images.githubusercontent.com/2810187/178842328-d979305a-6f0f-4a6b-87a4-47cea08eac1a.png)

</details>

## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- Windows 10
- In-game

## How to Test the Changes:


1. `.go c 67185`
2. Set the raid in normal 10 mode
3. Perform the band normally, even dying, to verify that the npc shows the gossip again when re-entering.

## Topics to test

- [ ] In each fight, a different gossip and text should be shown on the npc.
- [ ] The npc that triggers the event remains inactive until the previous event ends.
- [ ] The door closes at the start of the event (I have to reverse this, now I really like it much better and it makes sense this way, but it doesn't seem to work this way).
- [ ] The last fight could be invoked in 2 ways. Waiting on the platform for 60 seconds, or through the gossip. According to the video, the fight is only activated by the gossip, so the automatic start after 60 seconds is disabled.
- [ ] Upon death, please check that everything is back to normal.
- [ ] The translation of a final text of Tirion was missing, in fact, it was missing to relate it, because the translation already exists.
- [ ] The second boss, had a hardcore text, although I have to keep working on it, because as it has 2 texts, I have to randomly select one of them.

## Known Issues and TODO List:


- [x] Restructure the script, because although it works, it could be written in a different way, so as not to repeat so much code.
- [x] Move the IDS, both from the Gossip Menu and the Npc text, inside an enum.
- [x] Some npc text is also missing to be added to the gossip_menu. Actually, to link them. Although it works without that, the information must be in the database.
- [x] When the gossip is selected, the door should close to prevent the player from leaving the instance and the event from running normally. Currently, in some instances, it allows the player to exit even when the event has started.
- [x] The gossip flag should be available when the event ends in its entirety. To avoid that the player can start an event without finishing the previous one.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
